zeebe/broker/src/test/java/io/camunda/zeebe/broker/backup/ConcurrentBackupCompactionTest.java

```diff
@@ -0,0 +1,216 @@
+import io.atomix.primitive.partition.PartitionId;
+import io.atomix.primitive.partition.PartitionMetadata;
+import io.atomix.raft.impl.LogCompactor;
+import io.atomix.raft.metrics.RaftServiceMetrics;
+import io.atomix.raft.partition.RaftPartition;
+import io.atomix.raft.storage.log.RaftLog;
+import io.atomix.utils.concurrent.ThreadContext;
+import io.camunda.zeebe.backup.management.BackupService;
+import io.camunda.zeebe.broker.utils.InlineThreadContext;
+import io.camunda.zeebe.journal.JournalMetaStore;
+import io.camunda.zeebe.journal.file.SegmentedJournal;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.scheduler.SchedulingHints;
+import io.camunda.zeebe.snapshots.PersistedSnapshot;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStore;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStoreImpl;
+import io.camunda.zeebe.test.util.junit.AutoCloseResources.AutoCloseResource;
+import io.camunda.zeebe.util.FileUtil;
+import io.camunda.zeebe.util.buffer.DirectBufferWriter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ExtendWith(MockitoExtension.class)
+public class ConcurrentBackupCompactionTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ConcurrentBackupCompactionTest.class);
+  private static ActorScheduler actorScheduler;
+  private static final String SNAPSHOT_FILE_NAME = "file1";
+  @TempDir Path dataDirectory;
+  private SegmentedJournal journal;
+  @AutoCloseResource private FileBasedSnapshotStore snapshotStore;
+  @AutoCloseResource private InMemoryMockBackupStore backupStore;
+  @AutoCloseResource private BackupService backupService;
+  private final int nodeId = 1;
+  private final int partitionId = 1;
+  private LogCompactor logCompactor;
+  private final ThreadContext threadContext = new InlineThreadContext();
+  @Mock private RaftLog raftLog;
+  @Mock private RaftServiceMetrics raftMetrics;
+
+  @BeforeAll
+  static void beforeAll() {
+    actorScheduler = ActorScheduler.newActorScheduler().build();
+    actorScheduler.start();
+  }
+
+  @AfterAll
+  static void afterAll() throws ExecutionException, InterruptedException {
+    actorScheduler.stop().get();
```

Comment:
:wrench: I think `ActorScheduler` is `AutoCloseable`, so you could also just annotate it with `@AutoCloseResource` and remove the `afterAll` method entirely.

Reply:
Unfortunately `@AutoCloseResource` does not guarantee an order in which the resources are closed and the test fails to stop correctly, probably because the `ActorScheduler` is stopped before the last actor.

Reply:
I created a class `DynamicAutoCloseable` to extract the logic from `AutoCloseableRule` to be used without direct support from  Junit 4/5